web/config.py

```diff
@@ -860,10 +878,29 @@
 if os.path.exists(system_config_dir + '/config_system.py'):
     try:
         sys.path.insert(0, system_config_dir)
-        from config_system import *
+        import config_system
+        config_system_settings = get_variables_from_module('config_system')
+        user_config_settings.update(config_system_settings)
     except ImportError:
         pass
 
+# Update settings for 'LOG_FILE', 'SQLITE_PATH', 'SESSION_DB_PATH',
+# 'AZURE_CREDENTIAL_CACHE_DIR', 'KERBEROS_CCACHE_DIR', 'STORAGE_DIR'
+# of DATA_DIR is user defined
+data_dir_dependent_settings = ['LOG_FILE', 'SQLITE_PATH', 'SESSION_DB_PATH',
+                               'AZURE_CREDENTIAL_CACHE_DIR',
+                               'KERBEROS_CCACHE_DIR', 'STORAGE_DIR']
+
+if 'DATA_DIR' in user_config_settings:
+    for setting in data_dir_dependent_settings:
+        if setting not in user_config_settings:
+            data_dir = user_config_settings['DATA_DIR']
+            file_dir_name = config.__dict__[setting].split('/')[-1]
```

Comment:
Can you `locals()` to access local variables? Refer - https://docs.python.org/3/library/functions.html#locals.
I am worried `import config` inside config can create cyclic dependency.

Reply:
@yogeshmahajan-1903 I also noticed you've used `split('/')`. This will not work in case of Windows.

Reply:
Its better to use https://docs.python.org/3/library/os.path.html python OS lib for path manipulation. In your case - `os.path.basename` instead of split.

Reply:
Accommodated review comments